packages/admin/src/Resources/Resource.php

```diff
@@ -135,6 +135,10 @@ public static function can(string $action, ?Model $record = null): bool
             return true;
         }
 
+        if (method_exists($policy, 'before') && is_bool($condition = $policy->before($user, $action))) {
```

Comment:
What if the `before` method returns an authorization response instead of a boolean?

Reply:
We don't support those inside the method either ATM, so this is consistent for now

Reply:
Don't support in what way? I bet it still manages authorization correctly, but just doesn't throw the same status code again?

This is different though, since if `before` is **not** authorized, by returning a specific status code, we now actually **do** authorize since the `if` condition evaluates to `false` because of `is_bool`.

Reply:
Reverting until 3.x